src/coreclr/scripts/genEventing.py

```diff
@@ -345,37 +410,62 @@ def parseTemplateNodes(templateNodes):
 
     return allTemplates
 
-def generateClrallEvents(eventNodes,allTemplates, target_cpp, runtimeFlavor, write_xplatheader):
+def generateClrallEvents(eventNodes, allTemplates, target_cpp, runtimeFlavor, write_xplatheader, providerName, inclusionList, generatedFileType):
     clrallEvents = []
     for eventNode in eventNodes:
-        eventName    = eventNode.getAttribute('symbol')
+        eventName = eventNode.getAttribute('symbol')
+
+        if not includeEvent(inclusionList, providerName, eventName):
+            continue
+
         templateName = eventNode.getAttribute('template')
 
         #generate EventEnabled
-        if not target_cpp:
-            clrallEvents.append("static ")
-        clrallEvents.append("inline %s EventEnabled" % (getEventPipeDataTypeMapping(runtimeFlavor)["BOOL"]))
-        clrallEvents.append(eventName)
-        clrallEvents.append("(void) {return ")
-        clrallEvents.append("EventPipeEventEnabled" + eventName + "()")
-
-        if runtimeFlavor.coreclr or write_xplatheader:
-            if os.name == 'posix':
-                clrallEvents.append(" || (XplatEventLogger" +
-                ("::" if target_cpp else "_") +
-                "IsEventLoggingEnabled() && EventXplatEnabled" +
-                eventName + "());}\n\n")
+        if includeProvider (providerName, runtimeFlavor):
+            if not target_cpp:
+                clrallEvents.append("static ")
+            if not runtimeFlavor.nativeaot:
+                clrallEvents.append("inline ")
+            clrallEvents.append("%s EventEnabled" % (getEventPipeDataTypeMapping(runtimeFlavor)["BOOL"]))
+            clrallEvents.append(eventName)
+            clrallEvents.append("(void)")
+            if runtimeFlavor.nativeaot and (generatedFileType == "header" or generatedFileType == "source-impl-noop"):
```

Comment:
Can we just drop this condition? (Just switch on `generatedfileType` as in the code under this if).